src/Mod/BIM/ArchCurtainWall.py

```diff
@@ -279,16 +279,52 @@ def execute(self,obj):
             if not vdir.Length:
                 vdir = FreeCAD.Vector(0,0,1)
             vdir.normalize()
-            basevector = face.valueAt(fp[1],fp[3]).sub(face.valueAt(fp[0],fp[2]))
-            a = basevector.getAngle(vdir)
-            if (a <= math.pi/2+ANGLETOLERANCE) and (a >= math.pi/2-ANGLETOLERANCE):
-                facedir = True
-                vertsec = obj.VerticalSections
-                horizsec = obj.HorizontalSections
-            else:
-                facedir = False
-                vertsec = obj.HorizontalSections
-                horizsec = obj.VerticalSections
+
+            # Check if face if vertical in the first place
+            # Fix issue in 'Curtain wall vertical/horizontal mullion mix-up'
+            # https://github.com/FreeCAD/FreeCAD/issues/21845
+            #
+            p = face.findPlane()  # Curve face (surface) seems return no Plane
```

Comment:
```suggestion
            face_plane = face.findPlane()  # Curve face (surface) seems return no Plane
```

This is just a naming suggestion. There might be better, shorter options.

However, please consider not using single character variable names, unless they're for localized variables with a narrow scope (e.g. `i` for indices in a loop). I know the original code does use them, but for new code I would suggest to improve upon this practice.

Single-character variable names are in general difficult to trace (try to search for all occurrences of the "p" variable for instance) and understand (try to understand what the "p" variable does a few pages below where it's been defined). Because of the difficulty of searching and identifying the correct one, they're difficult and error-prone to refactor as well.

Reply:
> Single-character variable names are in general difficult to trace (try to search for all occurrences of the "p" variable for instance) and understand (try to understand what the "p" variable does a few pages below where it's been defined). Because of the difficulty of searching and identifying the correct one, they're difficult and error-prone to refactor as well.

Indeed, totally agreed; I have same experience when searching in gedit and same thought as you suggest, glad you point it out, as I am not trained as programmer, I am not very sure what might just be fundamental :)